lessons/githubPages/lesson.md

```diff
@@ -0,0 +1,226 @@
+      <th align="left">Publishing branch</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td align="left">User Pages site</td>
+      <td align="left"><code>username.github.io</code></td>
+      <td align="left">master</td>
+    </tr>
+    <tr>
+      <td align="left">Organization Pages site</td>
+      <td align="left"><code>orgname.github.io</code></td>
+      <td align="left">master</td>
+    </tr>
+    <tr>
+      <td align="left">Project Pages site owned by a <strong>user account</strong></td>
+      <td align="left"><code>username.github.io/projectname</code></td>
+      <td align="left">gh-pages</td>
+    </tr>
+    <tr>
+      <td align="left">Project Pages site owned by an <strong>organization</strong></td>
+      <td align="left"><code>orgname.github.io/projectname</code></td>
+      <td align="left">gh-pages</td>
+    </tr>
+  </tbody>
+</table>
+
+## Online Editing: Automatic Page Generator
+
+1. Create a new repository ([github.com/new](https://github.com/new))
+  - There are 4 types of Github Pages site (in the table above and on **[GitHub](https://help.github.com/articles/user-organization-and-project-pages/)**)
+  - **following naming conventions is really important!** `.github.io` for user and organizations
+
+2. Navigate to **Settings** and click **Automatic Page Generator**  
+Or just go to generated_pages url for your repository:  
+`https://github.com/<username>/<repository>/generated_pages/new`
+  - Make some edits to the content
+  - Select a theme/layout
+  - Publish!
+
+3. We have a website!  
+`<username/orgname>.github.io/<projectname>/`
+  - Let's check it out
+
+4. Make an update
+
+5. Check out the commit history
+
+## Working Locally: Git and Jekyll
+
+1. First, we need to make sure we have everything installed
+
+~~~
+$ git --version
+git version X.X.X
+$ jekyll -v
+jekyll X.X.X
+~~~
+
+2. We'll have to create (or choose) a repo for a project
+
+~~~
+$ git init first-website
+Initialized empty Git repository in <path/to/current/directory>/first-website/.git/
+~~~
+
+_Note: There are a few extra steps if you want a project page for an existing repository! Please see: [help.github.com/articles/creating-project-pages-manually/](https://help.github.com/articles/creating-project-pages-manually/)_
+
+### Jekyll Basics
+
+Jekyll is a package (ruby gem) that makes a `jekyll` executable available from your command line.
+
+There aren't too many commands to start off, let's test them out:
+
+`jekyll new`  
+`jekyll serve`
+
+and another one:  
+`jekyll build`  
+but when you use `serve` it calls build, so the command is not often needed
+
+~~~
+$ jekyll new first-website
+New jekyll site installed in <path/to/current/directory>/first-website.
+~~~
+
+So we have a project, let's check out what's there!
+
+~~~
+$ cd first-website
+$ ls -lFA
+$ git status
+~~~
+
+Lots of files. Before we dig in let's try to get our website built
+
+~~~
+$ jekyll serve
```

Comment:
Maybe they changed the default behaviour in a recent version update? Because `jekyll serve` didn't do that for me when I started using jekyll (about a year ago). Since then I've just always done `jekyll serve --watch`. Good to know I don't need to do that anymore!